sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -171,6 +171,26 @@ class Column(val expr: Expression) extends Logging {
     Column.fn(name, this, lit(other))
   }
 
+  /**
+   * A version of the `fn` method specifically designed for binary operations in PySpark
+   * that require logging information.
+   * This method is used when the operation involves another Column.
+   *
+   * @param name               The name of the operation to be performed.
+   * @param other              The value to be used in the operation, which will be converted to a
+   *                           Column if not already one.
+   * @param pysparkLoggingInfo A map containing logging information such as the fragment and
+   *                           call site from PySpark.
+   * @return A Column resulting from the operation.
+   */
+  private def fn(
+      name: String, other: Any, pysparkLoggingInfo: java.util.ArrayList[String]): Column = {
```

Comment:
shall we just take two string parameters instead of taking an ArrayList?

Reply:
Ah, two string parameters makes sense. Good suggestion, thanks!

Reply:
Updated!